chain/sub/incoming.go

```diff
@@ -618,3 +623,11 @@ func HandleIncomingMessages(ctx context.Context, mpool *messagepool.MessagePool,
 		// Do nothing... everything happens in validate
 	}
 }
+
+func recordFailure(ctx context.Context, metric *stats.Int64Measure, failureType string) {
+	ctx, _ = tag.New(
+		ctx,
+		tag.Upsert(metrics.FailureType, failureType),
+	)
+	stats.Record(ctx, metric.M(1))
+}
```

Comment:
We also have a local `recordFailure` inside the block validator, which may cause confusion.
Mind updating that code to use this function as well?
Also, go fmt please, there is a missing newline in the file.